### PR TITLE
[BugFix][C++] Add a FinalizeS3 to adapt S3 destroy of arrow 12.0.0

### LINKED ISSUE #184: [Bug]: "test_graph_info_load_from_s3" failed under build type of Release
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When I use `cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTS=ON ..` and then `make test`, I got an error:

> Start  6: test_graph_info_load_from_s3
> Test   6: test_graph_info_load_from_s3 ................***Exception: SegFault 10.66 sec

### Expected Behavior

pass this test

### Minimal Reproducible Example

```
cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTS=ON ..
make test
```

### Environment

- Operating system: macOs Ventura 13.3.1
- GraphAr version: v0.5.0

### Link to GraphAr Logs

_No response_

### Further Information

_No response_

### PR DESCRIPTION
**Pull request by acezen**
Signed-off-by: acezen <qiaozi.zwb@alibaba-inc.com>

Committed-by: acezen from Dev container

## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to GraphAr?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [CONTRIBUTING](https://github.com/alibaba/GraphAr/blob/main/CONTRIBUTING.rst) doc
- [ ] I have signed the CLA
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments
Fixed #184

### CODE REVIEW

**Comment by acezen** [comment]
The memory still get leak in Debug mode, checking

**Comment by acezen** [comment]
hi， @lixueclaire ，can you check the `test_graph_info_load_from_s3` on your local environment?

**Comment by lixueclaire** [comment]
> hi， @lixueclaire ，can you check the `test_graph_info_load_from_s3` on your local environment?

I have tried, but still got segmentation fault for this test under the build type of Release.

File: cpp/src/filesystem.cc [hunk 1]
```diff
@@ -13,6 +13,8 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+#include <iostream>
```
Comment (lixueclaire) [inline cpp/src/filesystem.cc:16]: Is this header file necessary?

**Comment by acezen** [comment]
Close the pull request since the arrow 12.0.0's workaround can not fix the aws sdk segmentfault. see #187

**Pull request closed without merging**